src/utils/daemon.js

```diff
@@ -106,20 +106,32 @@ function writeConfigFile (ipfsd, config) {
   fs.writeJsonSync(configPath(ipfsd), config, { spaces: 2 })
 }
 
-// Check for a "API.HTTPHeaders.Access-Control-Allow-Origin": ["*"]
+// Check for * and webui://- in allowed origins on API headers.
+// THe wildcard was a ipfsd-ctl default, that we dont want, and webui://- was an
+// earlier experiement that should be cleared out.
+//
+// We remove them the first time we find them. If we find it again on subsequent
+// runs then we leave them in, under the assumption that you really want it.
+// TODO: show warning in UI when wildcard is in the allowed origins.
 function checkCorsConfig (ipfsd) {
-  const updatedCorsConfig = store.get('updatedCorsConfig')
+  if (store.get('checkedCorsConfig')) {
```

Comment:
We shouldn't rely on stored values. What if you set up an IPFS node yourself and configure it yourself and then you start using IPFS Desktop with that repository? Will we change the configuration without the user knowing? Perhaps we should only do this on `ipfsd.init`.

Reply:
I am ok with that risk. I want to fix older config files for users who have installed ipfs-desktop already. The previous iteration of this PR was much more likely to only run for users who had installed desktop and hadn'd edited their API config block. But based on @lidel's feedback, it's now more proactive and will remove the wildcard even if you have added additional domains.

Reply:
Agreed. The end goal is to do this only on `init` of new repo, but for now we need to proactively fix security issue opened by default config in  `ipfsd`. 

It should  stay that way for a release or two to ensure all users  got the fix. Then we can make it more passive.